phoenix-core/src/main/java/org/apache/phoenix/hbase/index/write/ParallelWriterIndexCommitter.java

```diff
@@ -116,7 +117,10 @@ public void write(Multimap<HTableInterfaceReference, Mutation> toWrite) throws S
             // doing a complete copy over of all the index update for each table.
             final List<Mutation> mutations = kvBuilder.cloneIfNecessary((List<Mutation>)entry.getValue());
             final HTableInterfaceReference tableReference = entry.getKey();
-            final RegionCoprocessorEnvironment env = this.env;
+            if (env !=null && tableReference.getTableName().equals(
+                env.getRegion().getTableDesc().getNameAsString())) {
+                continue;
+            }
```

Comment:
Can we conditionally *not* filter our local index rows depending on the HBase version?